src/ui/SearchInterface/SearchInterface.ts

```diff
@@ -1046,7 +1046,8 @@ export class SearchInterface extends RootComponent implements IComponentBindings
 
   private get duplicatesFacets() {
     const duplicate = [];
-    const facets = ComponentsTypes.getAllFacetsInstance(this.root);
+    const facets = ComponentsTypes.getAllFacetsInstance(this.root).filter(cmp => cmp.type !== 'TimespanFacet');
```

Comment:
Hmm, maybe I am missing some context but this feels like a band-aid rather than a true fix. What is special about the `Timespan Facet` that makes it necessary to remove it from the array of facets?

Reply:
The TimespanFacet is not really a facet, it's only a facet wrapper. So it always comes out as a duplicate of it's FacetRange children. Also it doesn't possess the `dependsOn` option.

Reply:
I see. Wouldn't that mean the selector/logic we are using to retrieve the Range facets needs to be adjusted? Something along the lines of: `return range facets that are not inside a Timespan facet && Timespan facets`?

If there is a Timespan and an unrelated Range facet configured with the same field/id, or two Timespan facets configured with the same field/id, then would we catch that?

Reply:
As it is with this PR, let's say you have this:
```
 <div class="CoveoFacetRange" data-id="nice" data-tab="All" data-field="@rowid"></div>
 <div class="CoveoTimespanFacet" data-id="nice" data-tab="All"></div>
```
It will output a warning because you have two facets with the same id: the CoveoFacetRange (@rowid) and the CoveoFacetRange created by the CoveoTimespanFacet

Reply:
Warning:
```
<div class="CoveoFacetRange" data-tab="All" data-field="@date"></div>
<div class="CoveoTimespanFacet" data-tab="All"></div>
```
Warning:
```
<div class="CoveoTimespanFacet" data-tab="All"></div>
<div class="CoveoTimespanFacet" data-tab="All"></div>
```

Reply:
My first solution was to just remove Timespan facet from the `ComponentsTypes.getAllFacetsInstance` it seemed not to really break anything except for the "DropdownHeaderLabel"

Reply:
Ok, seems like it works. It would be nice to have unit tests for the cases you listed if it's not too complicated.

If it is, then a simpler test might be to check that Timespan facets are filtered off with the description or a comment in the test body giving context.

Reply:
This would be more of an integration test, but since it would only check for a warning it's a bit overkill.
AFAIK the dependsOn feature has already been tested in integration tests. I'll check if something could be done there.

Will add a comment explaining the situation.

Reply:
Just by reading the code relatively quickly, I think the flaw is instead in the `ComponentTypes.getAllFacetsElements` function, which search in the documents for all `.CoveoFacet .CoveoFacetRange .CoveoTimespanFacet` etc..

And since, as you said, a TimespanFacet is a wrapper of FacetRange  (and the FacetRange is thus included as a children of the TimeSpanFacet), both are returned by that call.

I think we should modify `ComponentTypes.getAllFacetsElements` so that it does not consider facets elements that are direct descendant of another facet.

It's not true that there are two facet with the same ID in the page, there is only one : TimespanFacet, it just creates a child FacetRange element.

And when we call a function that says "Give me all facets in the page", it should not consider "wrapped" ones.